src/SettingsView/SettingsView.js

```diff
@@ -32,14 +53,18 @@ class SettingsView extends ConnectionComponent {
 								showExpandableButton={true}/>
 							<CardText expandable={true}>
 								<Container>
-									<SearchBar fullWidth={true}/>
+									<SearchBar onInputChange={this.handleSearchInput}/>
 									{this.props.eventTypes.value.map((eventType) => {
-										return (<EventType
-											eventType={eventType}
-											key={eventType.Id}
-											deleteEventType={this.props.deleteEventType}
-											fullWidth="true"/>);
-									})}
+                                        if(this.searchMatches(eventType, this.state.searchText)) {

```

Comment:
I don't know, if it is better to not pass the searchText from the this.state here, the function could be accessing the state itself. The this context should already be available there since you binded it in the constructor.